crypto/fipsmodule/ec/ec_key.c

```diff
@@ -279,7 +279,14 @@ point_conversion_form_t EC_KEY_get_conv_form(const EC_KEY *key) {
 }
 
 void EC_KEY_set_conv_form(EC_KEY *key, point_conversion_form_t cform) {
+  if (key == NULL || key->group == NULL) {
+    OPENSSL_PUT_ERROR(EC, ERR_R_PASSED_NULL_PARAMETER);
+    return;
+  }
   key->conv_form = cform;
+  if (key->group->mutable_ec_group) {
```

Comment:
why do we set the conversion form in the parent EC_GROUP as well? Can that group be associated with other keys?

Reply:
Great question, I did this since OpenSSL was doing this in [their implementation](https://github.com/openssl/openssl/blob/12d14de641c299ec080edc521f7080acc44e366f/crypto/ec/ec_key.c#L916-L921) of `EC_KEY_set_conv_form`. The `EC_GROUP` associated with `EC_KEY` [is duplicated](https://github.com/aws/aws-lc/blob/ad93747dbb36d5c18c1e640e000e0d15d40ab513/crypto/fipsmodule/ec/ec_key.c#L208-L222) with `EC_GROUP_dup`, so I think the intention was that it wouldn't effect other keys.

I should move the check for `key->group` further below however.

Reply:
not sure I understand, more than one key can be associated with an `EC_GROUP`, right?

Reply:
Yes `EC_GROUP`s are used to generate multiple `EC_KEY`s based on the group parameters. However, each `EC_KEY` has it's own duplicated reference to `EC_GROUP` assigned to itself. `EC_KEY`'s pointer reference to `EC_GROUP` is now different from the original `EC_GROUP` pointer that was used. 
The contents of `EC_KEY->EC_GROUP` can be mutated without effecting the original pointer. The original `EC_GROUP` can be continued to be used to generate new `EC_KEY`s. 

It's a mess since we now have two structures saving the `conv_form`.. But now that we're doing this, I believe the intention was to allow `EC_KEY` and it's own `EC_GROUP` pointer's `conv_form` type to align. The original `EC_GROUP` pointer that was used to generate `EC_KEY` shouldn't be effected.

Reply:
Do you think adding a test for this would help? I'm open to doing so if it helps clarify the pointer boundaries :)

Reply:
oh ok, I got it now, looks good to me.

Reply:
Thanks, Samuel, for offering to add a test. I didn't review the PR, but I think adding a test that would explain to us (when we ask ourselves later) what is done here is a great idea, if not too much work. Could be in another PR if you prefer.

Reply:
No problem, I have subsequent PR for implementing this behavior for custom curves. I'll add the test in that PR since the same concept applies: https://github.com/aws/aws-lc/pull/1881